packages/database/src/core/PersistentConnection.ts

```diff
@@ -186,15 +195,55 @@ export class PersistentConnection extends ServerActions {
   }
 
   get(query: Query): Promise<string> {
-    const req: { [k: string]: unknown } = {
+    const deferred = new Deferred<string>();
+    const request = {
       p: query.path.toString(),
       q: query.queryObject()
     };
+    const outstandingGet = {
+      action: 'g',
+      request,
+      onComplete: (message: { [k: string]: unknown }) => {
+        const payload = message['d'] as string;
+        if (message['s'] === 'ok') {
+          this.onDataUpdate_(
+            request['p'],
+            payload,
+            /*isMerge*/ false,
+            /*tag*/ null
+          );
+          deferred.resolve(payload);
+        } else {
+          deferred.reject(payload);
+        }
+      }
+    };
+    this.outstandingGets_.push(outstandingGet);
+    this.outstandingGetCount_++;
+    const index = this.outstandingGets_.length - 1;
+
+    if (!this.connected_) {
+      const self = this;
```

Comment:
This is very old style JavaScript. If you use a lambda below (`() => {...}`) you won't need to rewrite `this` and will also remove the lint warning.